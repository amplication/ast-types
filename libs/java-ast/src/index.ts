// Core exports
export * from "./lib/core/AstNode";
export * from "./lib/core/Writer";

// AST nodes exports
export * from "./lib/ast/Access";
export * from "./lib/ast/Annotation";
export * from "./lib/ast/Class";
export * from "./lib/ast/ClassReference";
export * from "./lib/ast/CodeBlock";
export * from "./lib/ast/Enum";
export * from "./lib/ast/Field";
export * from "./lib/ast/GenericType";
export * from "./lib/ast/Import";
export * from "./lib/ast/Interface";
export * from "./lib/ast/Method";
export * from "./lib/ast/Parameter";
export * from "./lib/ast/Type";

// Utils exports
// (Add utility functions as needed)
