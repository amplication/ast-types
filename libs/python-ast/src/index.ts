export * from "./lib/ast";
export * from "./lib/core";
