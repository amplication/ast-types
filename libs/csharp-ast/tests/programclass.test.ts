import { CsharpSupport as csharp } from "../src";

describe("programclass", () => {
  it("program class full", () => {
    const myClass = csharp.programClass({
      references: [
        csharp.classReference({
          name: "MyClassName",
          namespace: "Amplication",
        }),
      ],
      startFileBlocks: [
        csharp.codeblock({
          code: 'Console.WriteLine(@"startFileBlocks");',
        }),
      ],
      builderServicesBlocks: [
        csharp.codeblock({
          code: 'Console.WriteLine(@"builderServicesBlocks");',
        }),
      ],
      appBlocks: [
        csharp.codeblock({
          code: 'Console.WriteLine(@"appBlocks");',
        }),
      ],
      catchBlocks: [
        csharp.codeblock({
          code: 'Console.WriteLine(@"catchBlocks");',
        }),
      ],
      finallyBlocks: [
        csharp.codeblock({
          code: 'Console.WriteLine(@"finallyBlocks");',
        }),
      ],
      endFileBlocks: [
        csharp.codeblock({
          code: 'Console.WriteLine(@"endFileBlocks");',
        }),
      ],
    });
    expect(myClass.toString()).toMatchSnapshot();
  });
  it("program class no finally", () => {
    const myClass = csharp.programClass({
      references: [
        csharp.classReference({
          name: "MyClassName",
          namespace: "Amplication",
        }),
      ],
      startFileBlocks: [
        csharp.codeblock({
          code: 'Console.WriteLine(@"startFileBlocks");',
        }),
      ],
      builderServicesBlocks: [
        csharp.codeblock({
          code: 'Console.WriteLine(@"builderServicesBlocks");',
        }),
      ],
      appBlocks: [
        csharp.codeblock({
          code: 'Console.WriteLine(@"appBlocks");',
        }),
      ],
      catchBlocks: [
        csharp.codeblock({
          code: 'Console.WriteLine(@"catchBlocks");',
        }),
      ],
      // finallyBlocks: ,
      endFileBlocks: [
        csharp.codeblock({
          code: 'Console.WriteLine(@"endFileBlocks");',
        }),
      ],
    });
    expect(myClass.toString()).toMatchSnapshot();
  });
  it("program no catch", () => {
    const myClass = csharp.programClass({
      references: [
        csharp.classReference({
          name: "MyClassName",
          namespace: "Amplication",
        }),
      ],
      startFileBlocks: [
        csharp.codeblock({
          code: 'Console.WriteLine(@"startFileBlocks");',
        }),
      ],
      builderServicesBlocks: [
        csharp.codeblock({
          code: 'Console.WriteLine(@"builderServicesBlocks");',
        }),
      ],
      appBlocks: [
        csharp.codeblock({
          code: 'Console.WriteLine(@"appBlocks");',
        }),
      ],
      finallyBlocks: [
        csharp.codeblock({
          code: 'Console.WriteLine(@"finallyBlocks");',
        }),
      ],
      endFileBlocks: [
        csharp.codeblock({
          code: 'Console.WriteLine(@"endFileBlocks");',
        }),
      ],
    });
    expect(myClass.toString()).toMatchSnapshot();
  });
  it("program no try catch", () => {
    const myClass = csharp.programClass({
      references: [
        csharp.classReference({
          name: "MyClassName",
          namespace: "Amplication",
        }),
      ],
      startFileBlocks: [
        csharp.codeblock({
          code: 'Console.WriteLine(@"startFileBlocks");',
        }),
      ],
      builderServicesBlocks: [
        csharp.codeblock({
          code: 'Console.WriteLine(@"builderServicesBlocks");',
        }),
      ],
      appBlocks: [
        csharp.codeblock({
          code: 'Console.WriteLine(@"appBlocks");',
        }),
      ],
      endFileBlocks: [
        csharp.codeblock({
          code: 'Console.WriteLine(@"endFileBlocks");',
        }),
      ],
    });
    expect(myClass.toString()).toMatchSnapshot();
  });
});
