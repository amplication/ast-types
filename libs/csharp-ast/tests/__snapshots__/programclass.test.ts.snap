// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`programclass program class full 1`] = `
"using Amplication;

Console.WriteLine(@"startFileBlocks");
try
{
    Console.WriteLine(@"builderServicesBlocks");
    Console.WriteLine(@"appBlocks");
    app.Run();
}
catch(Exception ex)
{
    Console.WriteLine(@"catchBlocks");
}
finally
{
    Console.WriteLine(@"finallyBlocks");
}
Console.WriteLine(@"endFileBlocks");
"
`;

exports[`programclass program class no finally 1`] = `
"using Amplication;

Console.WriteLine(@"startFileBlocks");
try
{
    Console.WriteLine(@"builderServicesBlocks");
    Console.WriteLine(@"appBlocks");
    app.Run();
}
catch(Exception ex)
{
    Console.WriteLine(@"catchBlocks");
}
Console.WriteLine(@"endFileBlocks");
"
`;

exports[`programclass program no catch 1`] = `
"using Amplication;

Console.WriteLine(@"startFileBlocks");
try
{
    Console.WriteLine(@"builderServicesBlocks");
    Console.WriteLine(@"appBlocks");
    app.Run();
}
finally
{
    Console.WriteLine(@"finallyBlocks");
}
Console.WriteLine(@"endFileBlocks");
"
`;

exports[`programclass program no try catch 1`] = `
"using Amplication;

Console.WriteLine(@"startFileBlocks");
Console.WriteLine(@"builderServicesBlocks");
Console.WriteLine(@"appBlocks");
app.Run();
Console.WriteLine(@"endFileBlocks");
"
`;
